### DISCUSSION TITLE
How to pass the webpart context to a CustomSuggestionProvider

### DISCUSSION THREAD

**Discussion by snowDVL** [microsoft-search/pnp-modern-search#1957] [Q&A]
I would like to load the Terms of a Termset as search box suggestions using the PnPJS library.
For that I need the context of the PnP search Box Webpart. How to get the context in a CustomSuggestionProvider?

**Comment by wobba** [comment]
You can follow https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_suggestions_provider/ and use pnpjs from there. And you should be able to follow e.g. https://pnp.github.io/pnpjs/concepts/authentication/#msal-spfx when you don't have the context available.

**Reply by snowDVL** [reply]
Thank you for your reply. If I understand correctly than there is no way to pass the context to a customsuggestionprovider and I shoud use msal to authenticate the customsuggestionprovider?

import { spfi, SPBrowser } from "@pnp/sp";
import { MSAL } from "@pnp/msaljsclient";
import "@pnp/sp/webs";

const sp = spfi("https://tenant.sharepoint.com/sites/dev").using(SPBrowser(), MSAL(configuration, authParams));

const webData = await sp.web();

SharePoint Framework is relatively new to me and I don't have a clue what configuration and authParams should look like. Some kind of tutorial on this topic would be very helpful. I always find authentication a complex topic to deal with. Anyway thanks for trying to point me in the right direction.

**Reply by snowDVL** [reply]
There must have been something wrong with my project. After I recreated my suggestionprovider the context was availabe and I was able to load the Terms of a Termset.

export class CustomSuggestionProvider extends BaseSuggestionProvider<ICustomSuggestionProviderProperties> {

    private _zeroTermSuggestions: ISuggestion[] = [];

    private _Terms: ISuggestion[] = [];
  
    public async onInit(): Promise<void> {

      sp.setup(this.context);  

      this._onSuggestionSelected.bind(this);
  
      this._getTerms()
    }

    private async _getTerms(){  
      const info: ITermInfo[]  = await sp.termStore.groups.getById("0ed5f1c2-91b8-4fea-a185-40099840fc3b") // This is Term Group ID  
      .sets.getById("0d52ee6f-17c1-4296-82e7-2ce7bb9f2eda").terms(); //this is  Term Set ID  
      console.log(JSON.stringify(info));
      info.forEach(i=>this._Terms.push({displayText: i.labels[0].name}));
    }  
...
